### PR TITLE
docs: add digital ocean docs

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

https://github.com/unjs/nitro/issues/50

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Adds docs for digital ocean preset.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/deploy/providers/digitalocean.md [hunk 1]
```diff
@@ -2,6 +2,31 @@
...
+   ```
+
+   [More information](https://docs.digitalocean.com/products/app-platform/how-to/use-environment-variables/).
+
+1. Finally, you will need to ensure you set an `engines.node` field in your app's `package.json` to ensure Digital Ocean uses Node 14+:
+
+   ```json
+   {
+      "engines": {
+         "node": "14.x"
```
Comment (pi0) [inline docs/deploy/providers/digitalocean.md:25]: Isn't 16 supported?

File: docs/deploy/providers/digitalocean.md [hunk 2]
```diff
@@ -2,6 +2,31 @@
...
+Nitro supports deploying on the [Digital Ocean App Platform](https://docs.digitalocean.com/products/app-platform/) with minimal configuration.
+
+## Set up application
+
+1. Create a new Digital Ocean app following the [guide](https://docs.digitalocean.com/products/app-platform/how-to/create-apps/).
+
+1. Next, you'll need to configure environment variables. In your app settings, ensure the following app-level environment variables are set:
+
+   ```bash
+   HOST=0.0.0.0
```
Comment (pi0) [inline docs/deploy/providers/digitalocean.md:14]: I believe we could include it in default node server

Comment (pi0) [reply docs/deploy/providers/digitalocean.md:14]: https://github.com/unjs/nitro/commit/b7c719349a0ee399a525937abb5905e50b0351f2

Comment (danielroe) [reply docs/deploy/providers/digitalocean.md:14]: You mean, listen on public interface by default?

**Pull request merged**